src/Components/Card/index.jsx

```diff
@@ -25,27 +25,36 @@ const Card = (props) => {
     });
   };
 
-  // Save text 
-  const saveChenges = () => {
-    const data = newState;
-    data.editModeFlag = false;
-    props.saveNewChanges(data);
-  };
-
   // Enable card editing mode
   const editModeOn = () => {
-    const data = newState;
-    data.editModeFlag = true;
-    data.cardSelectionFlag = false;
-    props.saveNewChanges(data);
+    props.onSaveChanges({
+      ...newState,
+      editModeFlag: true,
+      chooseСardFlag: false
+    });
+  }
+
+  // Save new text 
+  const saveChenges = () => {
+    props.onSaveChanges({
+      ...newState,
+      editModeFlag: false
+    });
   };
 
   // Undo all changes
   const abortChanges = () => {
-    const data = props.card;
-    data.editModeFlag = false;
-    props.saveNewChanges(data);
-  };
+    props.onSaveChanges({
+      ...props.card,
+      editModeFlag: false
+    });
+    setNewState({
+      ...props.card,
+      editModeFlag: false
+    })
+  }
+
+  const cls = props.card.chooseСardFlag === false ? 'card_is_green' : 'card_is_red';
```

Comment:
There is a wonderful package classNames that handlers conditional styling. Try it out, you'll possibly like it a lot:

https://www.npmjs.com/package/classnames